checkbox: Doesn't work on touch devices

narqo opened the pull request:
fix #1472

See ftlabs/fastclick#351 for reference

/cc @tadatuta @mishanga can't this changes bring some new sideeffects?

Commit:
checkbox: Doesn't work on touch devices

fix #1472

narqo commented:
It feels like the current problem is with our gemini tests for `checkbox`. [We're manipulating `.checkbox__control` explicitly](https://github.com/bem/bem-components/blob/2e8444a4ad7bf3a859748f377749d9fb65541ae1/gemini/checkbox.gemini.js#L26), but it feels like we need to use `.checkbox` as it looks like more fair way to iterate with the block.

I've checked test page with checkbox in Firefox (v35) and everything looks fine there. @tadatuta @sipayRT can you help me here?

sipayRT commented:
I'll check it

narqo commented:
@sipayRT any luck here?

sipayRT commented:
@narqo yes, I checked run tests on `.checkbox` instead of `.checkbox__control` - tests was written correctly because we can't get focused state on `<label>` tag. So `.checkbox__control` should be focused and `.checkbox` should be captured. I'm trying to fix gemini-errors now (but tests are running to slow now :( )

aristov commented:
@narqo ping

Commit:
Fix gemini tests for checkbox and checkbox-group

We should user `checkbox` block as a context of the suite if
possilbe, as it exactly what user uses in the browser.

See NOTE comment about `checkbox__control` and `focus`.

narqo commented on gemini/checkbox.gemini.js:
```diff
@@ -37,8 +37,9 @@ gemini.suite('checkbox', function(root) {
-                        actions.focus(this.checkbox);
+                    .capture('focused-hard', function(actions, find) {
+                        // NOTE: can't focus on `<label>`, so find `checkbox__control` explicitly
+                        actions.focus(find(element + '__control'));
```
@sipayRT @tadatuta i've fixed `gemini` tests to use `checkbox` instead of `checkbox__control`. The only problem is with with `focus()` element. But I'm sure, it's a gemini bug (gemini-testing/gemini#172)

narqo commented:
Finally, I've fixed gemini tests! But we still has a massive yastatic.net + hsts problems in all PRs.

/cc @tadatuta @sipayRT

narqo commented:
/cc @tadatuta passed! I'd be grateful if you could review it and add to v2.2.0
